validphys2/src/validphys/convolution.py

```diff
@@ -400,22 +423,24 @@ def appl(df):
     return sigma.groupby(level=0).apply(appl)
 
 
-def hadron_predictions(loaded_fk, pdf):
+def hadron_predictions(loaded_fk, pdf_list):
     """Implementation of :py:func:`fk_predictions` for hadronic observables."""
-    gv = functools.partial(evolution.grid_values, pdf=pdf)
-    res = _gv_hadron_predictions(loaded_fk, gv)
-    res.columns = range(pdf.get_members())
+    gv1 = functools.partial(evolution.grid_values, pdf=pdf_list[0])
+    gv2 = functools.partial(evolution.grid_values, pdf=pdf_list[1])
```

Comment:
This generates an ambiguity, when `pdf_list[0] == pdf_list[1]` it doesn't matter, but if they are different, whether one of the pdfs is just the central values can change things a lot.

I would add a check for that in `determine_pdfs` ( I put it in the template)